backstory/src/test/java/com/google/sps/perspective/PerspectiveAPIClientTest.java

```diff
@@ -0,0 +1,191 @@
+import au.com.origma.perspectiveapi.v1alpha1.models.Score;
+import com.google.sps.perspective.PerspectiveAPIClient;
+import com.google.sps.perspective.PerspectiveValues;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/** Quality tests for PerspectiveAPIClient */
+@RunWith(JUnit4.class)
+public final class PerspectiveAPIClientTest {
+
+  /** an instance of the PerspectiveAPIClient to use throughout our tests */
+  private static PerspectiveAPIClient client;
+  /** a mock PerspectiveAPI object to be used as the input api throughout the tests */
+  private static PerspectiveAPI mockAPI;
+  /** the array of attribute types that we want scores back for */
+  private static final AttributeType[] desiredTypes = { AttributeType.PROFANITY, AttributeType.TOXICITY, AttributeType.UNSUBSTANTIAL };;
+
+
+  @Before 
+  public void setUp() {
+    // will set up an API that only returns -1
+    mockAPI = createMockAPI(new HashMap<AttributeType, Float>()); 
+
+    client = new PerspectiveAPIClient(mockAPI);
+  }
+
+  /**
+   * Construct a PerspectiveAPIClient with a null PerspectiveAPI to ensure 
+   * that an IllegalArgumentException will be thrown.
+   */
+  @Test (expected = IllegalArgumentException.class)
+  public void nullAPIInput() {
+    client = new PerspectiveAPIClient(null); 
+  }
+
+  /**
+   * Call analyze() with a null Attribute[] to ensure 
+   * that an IllegalArgumentException will be thrown.
+   */
+  @Test (expected = IllegalArgumentException.class)
+  public void nullAttributeArrayInput() {
+    client.analyze(null, "foo");
+  }
+
+  /**
+   * Call analyze() with a null text input to ensure 
+   * that an IllegalArgumentException will be thrown.
+   */
+  @Test (expected = IllegalArgumentException.class)
+  public void nullTextInput() {
+    client.analyze(desiredTypes, null);
+  }
+
+  /**
+   * Call analyze() with an empty text input to ensure 
+   * that an IllegalArgumentException will be thrown.
+   */
+  @Test (expected = IllegalArgumentException.class)
+  public void emptyTextInput() {
+    client.analyze(desiredTypes, "");
+  }
+
+  /**
+   * Check that analyze() returns a PerspectiveValues object
+   * with a map of the correct size.
+   */
+  @Test
+  public void checkSizeOfMap() {
+    PerspectiveValues values = client.analyze(desiredTypes, "foo");
+
+    Assert.assertEquals(desiredTypes.length, values.getAttributeTypesToScores().size());
+  }
+
+  /**
+   * Check that analyze() returns a PerspectiveValues
+   * object with the same text as passed in.
+   */
+  @Test
+  public void checkTextSame() {
+    PerspectiveValues values = client.analyze(desiredTypes, "foo");
```

Comment:
Creating a local variable would make this input more obviously connected to the assertion:

```
String text = "foo";
PerspectiveValues values = client.analyze(desiredTypes, text);
Assert.assertEquals(text, values.getText());
```

Reply:
I agree. I think it might be even clearer just to have a default text constant throughout file. I think I'll add this:
`/** a piece of default text to use in tests */
  private static final String DEFAULT_TEXT = "foo";`

Reply:
Done